protocols/gossipsub/src/error.rs

```diff
@@ -22,7 +22,6 @@
 
 use libp2p_core::identity::error::SigningError;
 use libp2p_core::upgrade::ProtocolError;
-use prost_codec::Error as ProstCodecError;
```

Comment:
Style question: in Python we do that fairly often to disambiguate names. In rust the fully qualified name is preferred? (Just want to make sure my thinking is on the right track)

Reply:
It is more a style-preference. I prefer full-qualified names because:

- It saves an import line that is otherwise just noise.
- Common crate names are easy to remember. For example, I know that we are using a crate called `prost_codec` in this repository. If I come across a reference to `prost_codec::Error` anywhere in the source-code, my brain just goes "Ah yeah, that must be the[^1] error type of that crate".
- If a crate has an `Error` type, I can pretty much assume that it is the only `Error` type, otherwise the author would have named them differently (like `EncodeError` and `DecodeError` for example).

That is a lot of useful information to extract from a single line of code :)

On the contrary, if types are renamed on import:

- I have to interrupt my code reading flow if I come across `ProstCodecError`, go the the `use` statements and check, what it actually is.
- Not all programmers will rename it in the same pattern. Meaning I might come across a name that is unfamiliar to me, only to find out it is a type I actually know, just under a different name.

In the end, it is about efficiency in being able to understand code. Especially the last point can be quite annoying :)
When you are hunting a bug f.e. your brain is usually in "spot odd things"-mode and an unfamiliar name is "odd" but in the case of renaming types, it is usually a false-positive.

It is a style thing at the end of the day so YMMV :)

[^1]: Many crates by convention just have one error type.